test/e2e/servicedns.go

```diff
@@ -103,7 +104,7 @@ var _ = Describe("ServiceDNS", func() {
 
 		serviceDNS.Status = serviceDNSStatus
 		By("Waiting for the ServiceDNS object to have correct status")
-		common.WaitForObject(tl, namespace, serviceDNS.Name, objectGetter, serviceDNS, framework.PollInterval, framework.TestContext.SingleCallTimeout)
+		common.WaitForObject(tl, namespace, serviceDNS.Name, objectGetter, serviceDNS, framework.PollInterval, framework.TestContext.SingleCallTimeout, common.Equivalent)
```

Comment:
(No action required) Rather than requiring that every caller be updated to specify this argument that you want to vary at a new call site, consider adding a new function that the old function can call. This has the advantage of both minimizing code churn and minimizing the verbosity of calls that share arguments.

```golang
func WaitForObjectWithCheck(..., equivalentFunc func(...)) {...}

func WaitForObject(...) {
  WaitForObjectWithCheck(..., common.Equivalent)
}

```